### PR TITLE
fix: Fix DirEntryCache behavior

### PR DESCRIPTION
**Pull request by dmga44**
Our current dev branch is not passing our tests because it is failing on some Ganesha tests. This PR fixes those tests and also some  ShortSystemTests that were failing.

The errors that caused the various failures were introduced while merging the extend-dircache branch, therefore the fixes are related to its behavior.

### CODE REVIEW

**Review by lgsilva3087 — APPROVED** [review]

File: src/mount/direntry_cache.h [hunk 1]
```diff
@@ -474,6 +487,19 @@ class DirEntryCache {
 			++it;
 			erase(entry);
 		}
+
+		// Make sure inode_multiset_ is also clean of outdated entries.
+		// There are scenarios where we can not determine yet the inodes from
+		// the parent, like in unlink
```
Comment (Antuan-AE) [inline src/mount/direntry_cache.h:493]: Consider adding or extending existing Unit tests to checks this corner cases where entries are left hanging in cache and should be removed, or any other possible scenario we are not properly testing as of now...

File: src/mount/direntry_cache.h [hunk 2]
```diff
@@ -487,14 +513,27 @@ class DirEntryCache {
...
+		auto iter = inode_multiset_.lower_bound(parent_inode, InodeCompare());
+
+		while (iter != inode_multiset_.end() &&
+		       (iter->inode == parent_inode ||
+		        iter->parent_inode == kInvalidParent)) {
+			DirEntry *entry = std::addressof(*iter);
+			++iter;
+			erase(entry);
+		}
```
Comment (Antuan-AE) [inline src/mount/direntry_cache.h:536]: See my previous comment. Consider adding mocked data inside ```inode_multiset``` container to test this new implementation.

File: src/mount/direntry_cache.h [hunk 3]
```diff
@@ -573,17 +612,49 @@ class DirEntryCache {
 		return current_time_;
 	}
 
+	/// String representation of the complete cache. Useful for debugging.
```
Comment (Antuan-AE) [inline src/mount/direntry_cache.h:615]: Consider adding pre-processor directive like ```#ifdef DEBUG``` if this function is only needed for debugging to avoid adding this symbols to Release binaries.

**Review by Antuan-AE — APPROVED** [review]
LGTM.

File: src/mount/direntry_cache.h (hunk 1 above)
Comment (lgsilva3087) [reply src/mount/direntry_cache.h:493]: It will be addressed in this task:
https://3.basecamp.com/4375752/buckets/30999151/card_tables/cards/7271353671

File: src/mount/direntry_cache.h (hunk 2 above)
Comment (lgsilva3087) [reply src/mount/direntry_cache.h:536]: It will be addressed in this task:
https://3.basecamp.com/4375752/buckets/30999151/card_tables/cards/7271353671

File: src/mount/direntry_cache.h (hunk 3 above)
Comment (lgsilva3087) [reply src/mount/direntry_cache.h:615]: Will be done in later changes.

File: src/mount/direntry_cache.h [hunk 4]
```diff
@@ -263,14 +263,26 @@ class DirEntryCache {
 	 * \return True if inode has been found in cache, false otherwise.
 	 */
 	bool lookup(const SaunaClient::Context &ctx, uint32_t inode, Attributes &attr) {
+		// 'no more entries' marker
+		if (inode == 0) {
```
[the remark this answers is not in the document]
Comment (lgsilva3087) [reply src/mount/direntry_cache.h]: Done.
